rules/REQUEST-922-MULTIPART-ATTACK.conf

```diff
@@ -95,3 +95,26 @@ SecRule MULTIPART_PART_HEADERS "@rx content-transfer-encoding:(.*)" \
     ver:'OWASP_CRS/4.6.0-dev',\
     severity:'CRITICAL',\
     setvar:'tx.inbound_anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
+
+# Multipart header names can't contain any characters outside of range 33 and 126.
+# RFC 2045 refers RFC 822 about the header syntax.
+# Note: this is in phase:2 because these are headers that come in the body
+SecRule MULTIPART_PART_HEADERS "@rx [^\x20-\x7E]" \
```

Comment:
Let me ask if it is correct that the `x20` refers to the 32nd element in the ASCII table and the comment says that it should start from the 33rd.

Reply:
In my opinion, the relation between the comment and the rule is confusing. Either the comment should say:
> Multipart header names can't contain any characters outside of range 32 (space) and 126 (tilde).

or the rule should accept elements matching `[^\x21-\x7E]`.

Reply:
I agree. You're excluding the range `[0, 31]` and `[127,)`, but the comment says that you're excluding `[0, 32]` and `[127,)`.

```
# Multipart header names must only contain printable ASCII characters in the range of [33, 126], except colon.
# As `MULTIPART_PART_HEADERS` contains the full header line as a value, colon is still included in the match.
```

And the regex should be changed to what @huberts90 suggests. Nice find @huberts90!

Reply:
You're right, guys - I fixed it, see commit f377c6ac.